### ISSUE TITLE
source compatibility project violet fails with a compiler error

### ISSUE THREAD

**Issue by eeckstein** [swiftlang/swift-source-compat-suite#955]
### Description

```
/build/swift-source-compat-suite/project_cache/violet/Sources/Compiler/Implementation/CompilerImpl+Expr.swift:14:31: error: 'Expression' is ambiguous for type lookup in this context
 12 | extension CompilerImpl {
 13 | 
 14 |   internal func visit(_ node: Expression) throws {
    |                               `- error: 'Expression' is ambiguous for type lookup in this context
 15 |     self.setAppendLocation(node)
 16 |     try node.accept(self)
```

### Reproduction

builds:
https://ci.swift.org/job/swift-6.0-source-compat-suite-debug-ubuntu-2004/121/
https://ci.swift.org/job/swift-6.0-source-compat-suite-ubuntu-2004/121/
https://ci.swift.org/job/swift-source-compat-suite-debug-ubuntu-2004/499/
https://ci.swift.org/job/swift-source-compat-suite-ubuntu-2004/599/

log: https://ci.swift.org/job/swift-6.0-source-compat-suite-debug-ubuntu-2004/121/artifact/build/swift-source-compat-suite/FAIL_violet_5.3_BuildSwiftPackage.log

### Expected behavior

should build fine

### Environment

CI

**Comment by justice-adams-apple** [comment]
Hi @LiarPrincess , it appears the source compatibility suite has discovered an issue building [violet](https://github.com/LiarPrincess/Violet).

This will likely be uncovered in the next SDK update on our CI. Would you be interested in updating the violet source to account for this? If you find there is no longer any value in testing violet builds nightly, please let me know and I can remove it from the source compatability suite

**Comment by LiarPrincess** [comment]
Just a PR with an updated tag will suffice?

**Comment by justice-adams-apple** [comment]
>Just a PR with an updated tag will suffice?
Yep, or you can just send me an updated revision and I can open a PR quick and tag you for review
